Bluepill-cli/Bluepill-cli/Bluepill/Bluepill.m

```diff
@@ -406,7 +466,10 @@ - (void)deleteSimulatorWithContext:(BPExecutionContext *)context andStatus:(BPEx
         NEXT([self finishWithContext:context]);
         return;
     }
-
+    context.simulatorCreated = NO;  //also use this flag to tell writeSimUDIDFile() the simulator not avaiable
+    context.config.useSimUDID = nil; //prevent reuse this device when RETRY
+    self.config.useSimUDID = nil;
+    
```

Comment:
Previous comments from @vargon:
Does self.config need to be altered here? It should be treated as immutable. That's why we store executionConfigCopy which we do treat as mutable.

My comments:
Sorry I don't know self.config should be treated as immutable. But If we don't reset it here, the original value will be copied from self.config to context.config again when retry, and the invalid UDID may be reused, so only changing context.config is not enough. 

If we don't want to modify self.config in any way, one workaround is to add one more property to Bluepill class, e.g. noUseSimUDID, and use that to control if we should use useSimUDID, but that looks redundant and awkward too.